src/power_grid_model/core/power_grid_core.py

```diff
@@ -191,7 +191,7 @@ def cbind_func(self, *args, **kwargs):
         res = getattr(_CDLL, f"PGM_{name}")(*c_inputs)
         # convert to string for CStr
         if c_restype == CStr:
-            res = res.decode()
+            res = res.decode() if res is not None else ""
```

Comment:
so this is the bug

Reply:
Yep, the C API was behaving as intended but the Python wrapper wasn't